### ISSUE TITLE
Lucene not working correctly with "Cluster:xx" query

### ISSUE THREAD

**Issue by marcelalburg** [orientechnologies/orientdb#6594]
## OrientDB Version, operating system, or hardware.
- v2.2.7[x]
## Operating System
- [x] Linux
- [x] MacOSX
- [ ] Windows
- [ ] Other Unix
- [ ] Other, name? 
## Expected behavior and actual behavior

i've created an lucene index with following java:

```
final ODocument metadata = new ODocument();
    metadata.field("default", "org.apache.lucene.analysis.de.GermanAnalyzer");
regionType.createIndex(CLASS_NAME + ".name_de", "FULLTEXT", null, metadata, "LUCENE",
        new String[] { "name_de" });

```

i insert this test data:

`[{"name_de": "nord"},{"name_de":"nördlich"}]`
## Steps to reproduce the problem

My select query:

`select * from Region where [name_de] LUCENE 'nor*'`

returns the two rows and this is what i expect.

For the future, i'd like to query two or more classes with one query. i figure out, this is possible via "cluster:". my query is now:

`select * from cluster:Region where [name_de] LUCENE 'nor*'`

and this returns just one row -> "nord".

Is that a valid behaviour ?

Thanks
Marcel Alburg

**Comment by maggiolo00** [comment]
@marcelalburg 

What do you mean by

```
For the future, i'd like to query two or more classes with one query.
```

**Comment by marcelalburg** [comment]
Hey, sorry maybe i use the wrong expression to describe that.

in the future, i'll use something like this:

`select * from cluster:[Region,Abc,Cde] where [name_de] LUCENE 'nor*'`

Thanks
Marcel

**Comment by maggiolo00** [comment]
@marcelalburg 

I think it will not work.

Indexes works at class level not cluster level.

**Comment by marcelalburg** [comment]
Mhh ok, then i will use a manual index for this. 

Maybe you could answer me one more question. If i add multiple times the same value to the same vertex (#nn:n) to a manual index. is there any problem about that ?

Thanks for your answer
Marcel

**Comment by maggiolo00** [comment]
@marcelalburg 

no problem. You will get multiple times the same vertex in the results set.

Can you explain a little bit you use case maybe i can help more

Thanks

**Comment by StarpTech** [comment]
Hi @maggiolo00 indexes arent supported for Clusters? This would mean that the concept of clustering and distribution is uncomplete?! 

When I understand correct then clustering is a mechanism of grouping and the set are still classes.

**Comment by StarpTech** [comment]
The same issue is that `edges` cannot be deleted by a cluster selection.

```
DELETE EDGE SponoredBy CLUSTER S WHERE ....
```
Thats a workaround but here you can see that the clustering concept is very uncomplete.
```
delete edge EdgeClass where @rid in (select @rid from cluster:edge_cluster where .... )
```

**Comment by marcelalburg** [comment]
Hello,

i try to explain my usecase:

I've following classes: 

Continent
Country
Region
Area
aso.

Every Class has different properties but min. a "name" property.

and i like to create an AutoComplete/Suggest Lucene index on this field and all classes (this works). But i need to query them now.

I hope this explained it a little bit.

Thanks
Marcel

**Comment by marcelalburg** [comment]
Hello, i found an other solution for my problem.

i create a base Class "Searchable" wich one contains all fields for searching. On this class, i create my fulltext lucene index.

all other classes (see above) extends this "Searchable" class.

Now, it's possible to query just the "Searchable" class with the LUCENE Query.

Thanks
Marcel

**Comment by StarpTech** [comment]
Hi  @marcelalburg your solution is no answer to your issue. Its still an issue. Please reopen or I will open a new one because it will address me soon.

**Comment by StarpTech** [comment]
@maggiolo00 can you answer my questions? Thanks.

**Comment by robfrank** [comment]
Hi, on the next 2.2.16 version Lucene indexes will contain additional metadata fields **_CLUSTER** and **_CLASS**. So a query could be partitioned by cluster:

```
SELECT FROM Person WHERE name LUCENE '+_CLUSTER:26  OTHER QUERY ' "
```

**Comment by StarpTech** [comment]
Close the issue before it released. Awesome!

**Comment by robfrank** [comment]
What do you mean?

**Comment by StarpTech** [comment]
@robfrank the current release is 2.2.15 you closed the issue before we are able to test it.

**Comment by robfrank** [comment]
Can you test  with a snapshot? I mean 2.2.16-SNAPSHOT.

**Comment by robfrank** [comment]
this feature is included in 2.2.16. I close it, feel free to open a new issue relative to the feature as it is implemented.
